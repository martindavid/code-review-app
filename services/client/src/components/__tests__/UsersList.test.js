import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import UsersList from '../UsersList';

const users = [
  {
    'active': true,
    'email': 'mvalentino@martinlabs.me',
    'id': 1,
    'username': 'mvalentino',
  },
  {
    'active': true,
    'email': 'martin@gmail.com',
    'id': 2,
    'username': 'martin',
  }
];

test('UsersList render properly', () => {
  const wrapper = shallow(<UsersList users={users} />);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe('mvalentino');
})

test('UsersList renders a snapshot properly', () => {
  const tree = renderer.create(<UsersList users={users} />).toJSON();
  expect(tree).toMatchSnapshot();
})
