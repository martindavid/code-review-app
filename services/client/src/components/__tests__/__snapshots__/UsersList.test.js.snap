// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div>
  <h1
    className="title is-1"
  >
    All Users
  </h1>
  <hr />
  <br />
  <table
    className="table is-hoverable is-fullwidth"
  >
    <thead>
      <tr>
        <td>
          ID
        </td>
        <td>
          Email
        </td>
        <td>
          Username
        </td>
        <td>
          Active
        </td>
        <td>
          Admin
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          mvalentino@martinlabs.me
        </td>
        <td>
          mvalentino
        </td>
        <td>
          true
        </td>
        <td>
          false
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          martin@gmail.com
        </td>
        <td>
          martin
        </td>
        <td>
          true
        </td>
        <td>
          false
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
